### PR TITLE
Firmware and docs for running QMK on a (modded) Maltron DQz11N1G contoured keyboard

### PR DESCRIPTION
**Pull request by dvdkhlng**
## Description

This adds a new keyboard firmware: handwired/dqz11n1g that allows running QMK on a Maltron DQz11N1G ortholinear contoured keyboard when modded as described in the readme.md .

Note that I couldn't test the info.json file for correctness, as I run Linux exclusively and the QMK toolbox is not available here.  I also did not find a way to load a new info.json into the online QMK configurator.   Some help with that would be appreciated.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* none

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/handwired/dqz11n1g/rules.mk [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = caterina
+
+# Keyboard matrix uses shift-registers read via SPI
+CUSTOM_MATRIX = lite
+SRC += matrix.c spi_master.c
```
Comment (drashna) [inline keyboards/handwired/dqz11n1g/rules.mk:9]:
```suggestion
SRC += matrix.c 
QUANTUM_LIB_SRC += spi_master.c
```

**Review by drashna — CHANGES_REQUESTED** [review]
Also, images need to be rehosted on imgur or the like.

**Comment by dvdkhlng** [comment]
Hi @drashna , thanks for taking the time to review my changes.

changed rules.mk like you suggested.

> Also, images need to be rehosted on imgur or the like.

Can you confirm that hosting images externally to the QMK git repository, but still on github.com itself is not acceptable here?   Don't have an imgur account; right now the images are hosted in my github account's wiki.

**Comment by drashna** [comment]
> Can you confirm that hosting images externally to the QMK git repository, but still on github.com itself is not acceptable here? Don't have an imgur account; right now the images are hosted in my github account's wiki.

Correct.  

https://docs.qmk.fm/#/pr_checklist?id=keyboard-prs
> images should be uploaded to an external image hosting service, such as [imgur](https://imgur.com/).

There are a number of reasons for this, but one of them is, if you close your github account, it removes the images.

File: keyboards/handwired/dqz11n1g/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+enum layer_names { _DEFAULT, _FUNCTION };
+#define KC_FUN TG(_FUNCTION)
+#define KC_WWW KC_WWW_SEARCH
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [_DEFAULT] = LAYOUT_dqz11n1g(
```
Comment (noroadsleft) [inline keyboards/handwired/dqz11n1g/keymaps/default/keymap.c:23]:
```suggestion
    [_DEFAULT] = LAYOUT(
```

**Comment by dvdkhlng** [comment]
@noroadsleft : applied your changes verbatim and just retested firmware, sill works as before.  Note that I couldn't test the info.json changes myself (qmk toolbox not available under Linux).

**Comment by noroadsleft** [comment]
@dvdkhlng, `info.json` is my specialty when it comes to QMK. :)

[This change](https://github.com/qmk/qmk_firmware/pull/17237#discussion_r921356571) is because I had you change the layout macro name in `dqz11n1g.h` to `LAYOUT` – this reference has to match the layout macro name exactly for the data in `info.json` to work as intended.

[This change](https://github.com/qmk/qmk_firmware/pull/17237#discussion_r921356820) is to give the file the correct syntax. This particular section is an array type, which contains objects that are comma-separated. In JSON, the last item in an array should not have a trailing space. QMK Configurator's API is the main consumer of `info.json` data, and will ignore the trailing space, but there are other tools in QMK that will refuse to parse the file because of the invalid syntax.

QMK Toolbox actually doesn't use `info.json` at all.

**Review by noroadsleft — APPROVED** [review]

**Comment by noroadsleft** [comment]
Thanks!

**Comment by dvdkhlng** [comment]
> Thanks!
@noroadsleft , thank you too; this finally upgrades my keyboard from being a DiY hack to "upstream QMK" :)
Yes, the role info.json was not too obvious when reading through the makefiles, but seeing now how it allows me to look and edit the layout via the [online configurator](https://config.qmk.fm/#/handwired/dqz11n1g/LAYOUT), is quite fantastic to say the least.

**Pull request merged**